hook.c

```diff
@@ -460,24 +454,20 @@ void mutt_message_hook(struct Mailbox *m, struct Email *e, HookFlags type)
       if ((mutt_pattern_exec(SLIST_FIRST(hook->pattern), 0, m, e, &cache) > 0) ^
           hook->regex.not)
       {
-        if (mutt_parse_rc_line(hook->command, &token, &err) == MUTT_CMD_ERROR)
+        if (mutt_parse_rc_line(hook->command, token, err) == MUTT_CMD_ERROR)
         {
-          FREE(&token.data);
-          mutt_error("%s", err.data);
-          current_hook_type = 0;
-          FREE(&err.data);
-
-          return;
+          mutt_error("%s", mutt_b2s(err));
+          break;
```

Comment:
[0] Correct me if I'm wrong: Changing the `return` to the `break`changes the behavior of the method by continuing on error instead of bailing.

If so, we should extract the `return` to `break` change to its own small commit. It's out of scope of a tidying commit, which shouldn't change behavior.

Reply:
Yes and no.
It's cheeky of me to mix this in, but the function's behaviour is unchanged
(with the possible exception below where I've messed up the return code).

I'll refactor this commit into the `Buffer` changes and the control-flow changes.